lametro/models/cms.py

```diff
@@ -276,23 +281,36 @@ class FiscalYearCalendar(models.Model):
         on_delete=models.SET_NULL,
         related_name="+",
     )
+    cal_type = models.CharField(
+        choices=TYPE_CHOICES, max_length=20, null=True, verbose_name="Calendar Type"
+    )
 
     panels = [
         FieldPanel("title"),
         FieldPanel("calendar"),
+        FieldPanel("cal_type"),
     ]
 
     def __str__(self):
         return self.title
 
     def clean(self):
-        # Ensure that only one calendar exists at a time
+        # Ensure that only one calendar for each type exists at a time
         model = self.__class__
-        if model.objects.count() > 0 and self.id != model.objects.get().id:
+        num_types = len(self.TYPE_CHOICES)
+        is_new_cal = not model.objects.filter(id=self.id).exists()
+
+        if model.objects.count() >= num_types and is_new_cal:
             raise ValidationError(
-                "Only one calendar can exist at a time. "
-                "Please edit the existing calendar object."
+                f"Only {num_types} calendars can exist at a time. "
+                "Please edit the existing calendars."
             )
```

Comment:
I figure that basing this first conditional off of the number of possible calendar types would make it easier to just drop in new types if/when they want another.